src/main/scala/cromwell/engine/WorkflowDescriptor.scala

```diff
@@ -0,0 +1,205 @@
+package cromwell.engine
+
+import java.nio.file.{Path, Paths}
+
+import ch.qos.logback.classic.encoder.PatternLayoutEncoder
+import ch.qos.logback.classic.spi.ILoggingEvent
+import ch.qos.logback.classic.{Level, LoggerContext}
+import ch.qos.logback.core.FileAppender
+import com.typesafe.config.{Config, ConfigFactory}
+import cromwell.binding._
+import cromwell.engine.backend.Backend
+import cromwell.engine.workflow.WorkflowOptions
+import cromwell.server.CromwellServer
+import lenthall.config.ScalaConfig._
+import org.slf4j.helpers.NOPLogger
+import org.slf4j.{Logger, LoggerFactory}
+import spray.json.{JsObject, _}
+
+import scala.util.{Failure, Success, Try}
+import scalaz.Scalaz._
+
+case class WorkflowDescriptor(id: WorkflowId,
+                              sourceFiles: WorkflowSourceFiles,
+                              workflowOptions: WorkflowOptions,
+                              rawInputs: Map[String, JsValue],
+                              namespace: NamespaceWithWorkflow,
+                              coercedInputs: WorkflowCoercedInputs,
+                              declarations: WorkflowCoercedInputs,
+                              backend: Backend,
+                              configCallCaching: Boolean,
+                              lookupDockerHash: Boolean) {
+  import WorkflowDescriptor._
+
+  val shortId = id.toString.split("-")(0)
+  val ioInterface = backend.ioInterface(workflowOptions)
+  val name = namespace.workflow.unqualifiedName
+  val actualInputs: WorkflowCoercedInputs = coercedInputs ++ declarations
+  val props = sys.props
+
+  private lazy val optionCacheWriting = workflowOptions.getBoolean("write_to_cache") getOrElse configCallCaching
+  private lazy val optionCacheReading = workflowOptions.getBoolean("read_from_cache") getOrElse configCallCaching
+
+  if (!configCallCaching) {
+    if (optionCacheWriting) logWriteDisabled()
+    if (optionCacheReading) logReadDisabled()
+  }
+
+  lazy val writeToCache = configCallCaching && optionCacheWriting
+  lazy val readFromCache = configCallCaching && optionCacheReading
+
+  lazy val workflowLogger = props.get("LOG_MODE") match {
+    case Some(x) if x.toUpperCase.contains("SERVER") => makeFileLogger(
+      Paths.get(props.getOrElse("LOG_ROOT", ".")),
+      s"workflow.$id.log",
+      Level.toLevel(props.getOrElse("LOG_LEVEL", "debug"))
+    )
+    case _ => NOPLogger.NOP_LOGGER
+  }
+
+
+
```

Comment:
well I guess if you take out a bunch of blank lines you need to put new ones back in... :stuck_out_tongue: